src/views.py

```diff
@@ -57,11 +58,45 @@ def get_classrooms(request: HttpRequest) -> HttpResponse:
     for classroom in classrooms:
         building = classroom["building"]
         room = classroom["room"]
-        start_time = ScheduleBlock.objects.filter(room=room).values("start_time")[0]
-        end_time = ScheduleBlock.objects.filter(room=room).values("end_time")[0]
+        day_of_the_week = DAY_OF_THE_WEEK_MAPPING[int(data["day"])]
+
+        start_times = (
+            ScheduleBlock.objects.filter(
+                building=building,
+                room=room,
+                day_of_the_week=day_of_the_week,
+                start_time__gte=data["end_time"],
+            )
+            .order_by("start_time")
+            .values("start_time")
+        )
+
+        end_times = (
+            ScheduleBlock.objects.filter(
+                building=building,
+                room=room,
+                day_of_the_week=day_of_the_week,
+                end_time__lte=data["start_time"],
+            )
+            .order_by("-end_time")
+            .values("start_time")
+        )
 
         # string to display room, start time, and end time
-        room_time_str = f"Room: {room}" + " Start time: " + start_time['start_time'].strftime("%I:%M %p") + " End Time: " + end_time['end_time'].strftime("%I:%M: %p")
+        room_time_str = (
+            f"Room: {room} (Open from "
+            + (
+                start_times[0]["start_time"].strftime("%I:%M %p")
+                if len(start_times) > 0
+                else " beginning of day"
+            )
+            + " - "
+            + (
+                end_times[0]["start_time"].strftime("%I:%M %p") + ")"
+                if len(end_times) > 0
+                else " end of day)"
+            )
+        )
```

Comment:
The if else is to return a default "open from beginning of day or until end of day" if we don't find any blocks before/after